### PR TITLE
New secured API keys

### PR DESCRIPTION
**Pull request by JanPetr**

### CODE REVIEW

File: app/design/frontend/base/default/template/algoliasearch/internals/configuration.phtml [hunk 1]
```diff
@@ -139,9 +139,36 @@ if ($config->isInstantEnabled() && $isSearchPage) {
...
+/** Generate API keys */
+$instantsearchApiKeyParams = $autocompleteApiKeyParams = $config->getAttributesToRetrieve($customerGroupId);
+
+/**
+ * Add custom instantsearch filter restrictions into $instantsearchCustomFilters variable
+ * Docs: https://www.algolia.com/doc/api-client/php/api-keys#generate-key
+ */
+$instantsearchCustomFilters = '';
+
+$instantsearchApiKeyParams['filters'] = $instantsearchCustomFilters.'visibility_search=1';
```
Comment (maxiloc) [inline app/design/frontend/base/default/template/algoliasearch/internals/configuration.phtml]: I would use facetFilters, the syntax will be clearer, and we let filters empty for people wanting to filter on top of that

Comment (JanPetr) [inline app/design/frontend/base/default/template/algoliasearch/internals/configuration.phtml]: Makes sense. Changed ;)

File: app/design/frontend/base/default/template/algoliasearch/internals/configuration.phtml [hunk 2]
```diff
@@ -139,9 +139,41 @@ if ($config->isInstantEnabled() && $isSearchPage) {
...
+$attributesToRetrieve = $config->getAttributesToRetrieve($customerGroupId);
+
+if ($attributesToRetrieve) {
+    $instantsearchApiKeyParams['attributesToRetrieve'] = $attributesToRetrieve;
+    $autocompleteApiKeyParams['attributesToRetrieve'] = $attributesToRetrieve;
+}
+
+/**
+ * Add custom instantsearch filter restrictions into $instantsearchApiKeyParams['filters'] variable
+ * Docs: https://www.algolia.com/doc/api-client/php/api-keys#generate-key
```
Comment (maxiloc) [inline app/design/frontend/base/default/template/algoliasearch/internals/configuration.phtml]: I would put this doc link next to generateSearchSecuredApiKey call and put there the link to the filter param doc + explain 1 or 2 usecases

Comment (JanPetr) [inline app/design/frontend/base/default/template/algoliasearch/internals/configuration.phtml]: 👍 Take a look ;)

Comment (maxiloc) [inline app/design/frontend/base/default/template/algoliasearch/internals/configuration.phtml]: Yes better, I think we should mention that "Putting a filter in the key will make it non clearable by instantsearch.js"

Comment (JanPetr) [inline app/design/frontend/base/default/template/algoliasearch/internals/configuration.phtml]: Not sure, I'd find it more confusing with the "clearable" sentence

Comment (maxiloc) [inline app/design/frontend/base/default/template/algoliasearch/internals/configuration.phtml]: My point is looking at the code right now I don't get why I would use it. I agree the wording I use is not good 😄

**Pull request merged**